arch/posix/core/fatal.c

```diff
@@ -34,41 +37,40 @@ const NANO_ESF _default_esf = {
 FUNC_NORETURN void _NanoFatalErrorHandler(unsigned int reason,
 		const NANO_ESF *esf)
 {
-#ifdef CONFIG_PRINTK
+	LOG_PANIC();
+
 	switch (reason) {
 	case _NANO_ERR_CPU_EXCEPTION:
 	case _NANO_ERR_SPURIOUS_INT:
 		break;
 
 	case _NANO_ERR_INVALID_TASK_EXIT:
-		printk("***** Invalid Exit Software Error! *****\n");
+		LOG_ERR("***** Invalid Exit Software Error! *****");
 		break;
 
 
 	case _NANO_ERR_ALLOCATION_FAIL:
-		printk("**** Kernel Allocation Failure! ****\n");
+		LOG_ERR("**** Kernel Allocation Failure! ****");
```

Comment:
By the way, there is no backend for the posix arch yet. So as of now, this changes cannot be tested.